File: src/main/java/org/oscarehr/integration/myhealthaccess/model/MhaPatientAccess.java

/**
 * Copyright (c) 2012-2018. CloudPractice Inc. All Rights Reserved.
 * This software is published under the GPL GNU General Public License.
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 *
 * This software was written for
 * CloudPractice Inc.
 * Victoria, British Columbia
 * Canada
 */
 
package org.oscarehr.integration.myhealthaccess.model;

import lombok.Data;

import java.time.ZonedDateTime;

@Data
public class MhaPatientAccess
{
	// core information
	protected String patientId;
	protected String clinicId;
	protected MHAPatient.LINK_STATUS linkStatus;

	// user permissions
	protected Boolean canMessage;
	protected Boolean canCancelAppointments;

	// detail connection info
	// confirmation
	protected ZonedDateTime confirmedAt;
	protected String confirmedById;
	protected MhaUserType confirmedByType;
	protected String confirmingUserName;
	// verification
	protected ZonedDateTime verifiedAt;
	protected String verifiedById;
	protected MhaUserType verifiedByType;
	protected String verifierUserName;
}

File: src/main/frontend/src/lib/integration/myhealthaccess/model/MhaPatientAccess.ts
import {LinkStatus} from "./LinkStatus";
import {Moment} from "moment";
import {MhaUserType} from "./MhaUserType";

export default class MhaPatientAccess
{
	// core
	protected _patientId: string;
	protected _clinicId: string;
	protected _linkStatus: LinkStatus;

	// user permissions
	protected _canMessage: boolean;
	protected _canCancelAppointments: boolean;

	// confirmation
	protected _confirmedAt: Moment;
	protected _confirmedById: string;
	protected _confirmedByType: MhaUserType;
	protected _confirmingUserName: string;

	// verification
	protected _verifiedAt: Moment;
	protected _verifiedById: string;
	protected _verifiedByType: MhaUserType;
	protected _verifierUserName: string;

	// ==========================================================================
	// Setters
	// ==========================================================================

	set patientId(value: string)
	{
		this._patientId = value;
	}

	set clinicId(value: string)
	{
		this._clinicId = value;
	}

	set linkStatus(value: LinkStatus)
	{
		this._linkStatus = value;
	}

	set canMessage(value: boolean)
	{
		this._canMessage = value;
	}

	set canCancelAppointments(value: boolean)
	{
		this._canCancelAppointments = value;
	}

	set confirmedAt(value: Moment)
	{
		this._confirmedAt = value;
	}

	set confirmedById(value: string)
	{
		this._confirmedById = value;
	}

	set confirmedByType(value: MhaUserType)
	{
		this._confirmedByType = value;
	}

	set confirmingUserName(value: string)
	{
		this._confirmingUserName = value;
	}

	set verifiedAt(value: Moment)
	{
		this._verifiedAt = value;
	}

	set verifiedById(value: string)
	{
		this._verifiedById = value;
	}

	set verifiedByType(value: MhaUserType)
	{
		this._verifiedByType = value;
	}

	set verifierUserName(value: string)
	{
		this._verifierUserName = value;
	}

	// ==========================================================================
	// Getters
	// ==========================================================================

	get patientId(): string
	{
		return this._patientId;
	}

	get clinicId(): string
	{
		return this._clinicId;
	}

	get linkStatus(): LinkStatus
	{
		return this._linkStatus;
	}

	get canMessage(): boolean
	{
		return this._canMessage;
	}

	get canCancelAppointments(): boolean
	{
		return this._canCancelAppointments;
	}

	get confirmedAt(): Moment
	{
		return this._confirmedAt;
	}

	get confirmedById(): string
	{
		return this._confirmedById;
	}

	get confirmedByType(): MhaUserType
	{
		return this._confirmedByType;
	}

	get confirmingUserName(): string
	{
		return this._confirmingUserName;
	}

	get verifiedAt(): Moment
	{
		return this._verifiedAt;
	}

	get verifiedById(): string
	{
		return this._verifiedById;
	}

	get verifiedByType(): MhaUserType
	{
		return this._verifiedByType;
	}

	get verifierUserName(): string
	{
		return this._verifierUserName;
	}
}